extensions/qcom/GLSL_QCOM_image_processing.txt

```diff
@@ -0,0 +1,231 @@
+    |       sampler2D tex,               | a 2D kernel of filter weights with a corr- |
+    |       vec2      P,                 | esponding region of sampled texels and     |
+    |       sampler2DArray weight)       | sums the results to produce the output     |
+    |                                    | value.                                     |
+    +------------------------------------+--------------------------------------------+
+    |   vec4 textureBoxFilterQCOM(       | Linear operation taking average of pixels  |
+    |       sampler2D tex,               | within the spatial region described by     |
+    |       vec2       P,                | boxSize.  The box is centered at coordinate|
+    |       vec2       boxSize)          | P and has width and height of boxSize.x    |
+    |                                    | and boxSize.y.                             |
+    +------------------------------------+--------------------------------------------+
+    |   vec4 textureBlockMatchSADQCOM(   | Block matching operation measures the      |
+    |       sampler2D target             | correlation (or similarity) of the target  |
+    |       uvec2     targetCoord,       | block and reference block.  targetCoord    |
+    |       sampler2D reference,         | and refCoord specify the bottom-left corner|
+    |       uvec2     refCoord,          | of the block in target and reference       |
+    |       uvec2     blockSize)         | images. The error metric is the Sum of     |
+    |                                    | Absolute Differences (SAD).                |
+    +------------------------------------+--------------------------------------------+
+    |   vec4 textureBlockMatchSSDQCOM(   | Block matching operation measures the      |
+    |       sampler2D target             | correlation (or similarity) of the target  |
+    |       uvec2     targetCoord,       | block and reference block.  TargetCoord    |
+    |       sampler2D reference,         | and refCoord specify the bottom-left corner|
+    |       uvec2     refCoord,          | of the block in target and reference       |
+    |       uvec2     blockSize)         | images. The error metric is the Sum of     |
+    |                                    | Square Differences (SSD).                  |
+    +------------------------------------+--------------------------------------------+
+
+    * target and reference specify the two input textures. If
+    the same block from one texture is going to be used repeatedly across multiple
+    instructions, that texture should be specified as the referenceTexture.
+
+    * targetCoord and referenceCoord specify the texture coordinates (in texels)
+    of the bottom-left corner of each block.
+
+    * blockSize specifies the size (in texels) of the blocks to be compared.
+
+
+Modifications to the OpenGL ES Shading Language Specification, Version 3.10, dated 29 January 2016
```

Comment:
Are these the same additions for ES and desktop? It seems a shame to write the whole thing out twice. Perhaps just change this section to note any differences between the ES and desktop versions?